book/src/proposals/leader-duplicate-block-slashing.md

```diff
@@ -0,0 +1,219 @@
+# Leader Duplicate Block Slashing
+
+This design describes how the cluster slashes leaders that produce
+duplicate blocks.
+
+Leaders that produce multiple blocks for the same slot increase the
+number of potential forks that the cluster has to resolve.
+
+## Shred Format
+
+Shreds are produced by leaders during their scheduled slot.  Each
+shred is signed by the leader and is transmitted to the cluster via
+Turbine. A shred contains the following
+
+* signature
+* header: slot index, shred index
+* message data
+
+The signature is of the merkle tree of the shred data.
+
+```text
+              merke root
+    /                        \
+(slot index, shred index)        hash(msg)
+```
+
+## Proof of a Duplicate Shred
+
+Any two different signatures that show a merkle path to the same
+`(slot index, shred index)` for the same leader are proof that the
+leader produced two conflicting blocks.
+
+## Storing Multiple Versions of the Same Slot
+When there is a possibility of duplicate blocks, a validator need to track and
+replay multiple versions of a slot in case any one of these versions is 
+finalized by the cluster.
+
+### Generating a Unique Identifier for Each Version of a Slot
+We distinguish different versions of a slot by the ending blockhash. 
+This blockhash thus needs to be "functionally unique" (meaning a malicious
+leader cannot grind out a duplicate in the amount of time they have to
+generate a block) across all possible shred outputs generated by a leader. 
+
+To this end it's important to note that:
+1) Two different versions of a slot must generate a different set of shreds.
+Two different versions of a slot must either:
+    a) Chain from different parents, in which case the PoH of the entries will 
+    different
+    b) Contain different set of transactions, in which case the PoH of the 
+    entries will be different
+
+2) The resulting merkle trees of two different sets of shreds have 
+"functionally unique" merkle roots.
+
+Then if we were to generate a merkle tree out of the shred batches for a slot, 
+and define the blockhash of a slot to be:
+
+ `hash(last tick, shred merkle root)`
+
+This blockhash should be "functionally unique".
+
+### Generating the Merkle Tree for the Shreds
+1) Leader blocks on Poh before last tick can be generated
+2) Leader waits for BroadcastStage to finish generating all the shreds
```

Comment:
no need to name BroadcastStage, suffices to say that all shreds for a slot must be generated before step 3)